packages/cli-clean/src/clean.ts

```diff
@@ -0,0 +1,216 @@
+    const filename = path.resolve(startPath, file);
+    if (fileExists(filename)) {
+      return filename;
+    }
+  }
+
+  return undefined;
+}
+
+async function promptForCaches(
+  commands: CLICommand,
+): Promise<string[] | undefined> {
+  const names: string[] = Object.keys(commands);
+  const {caches} = await prompts({
+    type: 'multiselect',
+    name: 'caches',
+    message: 'Select all caches to clean',
+    choices: names.map((cmd) => ({
+      title: cmd,
+      value: cmd,
+      selected: DEFAULT_CATEGORIES.includes(cmd),
+    })),
+    min: 1,
+  });
+  return caches;
+}
+
+export async function clean(
+  _argv: string[],
+  _config: CLIConfig,
+  cleanOptions: Args,
+): Promise<void> {
+  const {include, projectRoot, verifyCache} = cleanOptions;
+  if (!fileExists(projectRoot)) {
+    throw new Error(`Invalid path provided! ${projectRoot}`);
+  }
+
+  const COMMANDS: CLICommand = {
+    android: [
+      {
+        label: 'Clean Gradle cache',
+        action: async () => {
+          const candidates =
+            os.platform() === 'win32'
+              ? ['android/gradlew.bat', 'gradlew.bat']
+              : ['android/gradlew', 'gradlew'];
+          const gradlew = findPath(projectRoot, candidates);
+          if (gradlew) {
+            const script = path.basename(gradlew);
+            await execa(
+              os.platform() === 'win32' ? script : `./${script}`,
+              ['clean'],
+              {cwd: path.dirname(gradlew)},
+            );
+          }
+        },
+      },
+    ],
+    cocoapods: [
+      {
+        label: 'Clean CocoaPods cache',
+        action: async () => {
+          await execa('pod', ['cache', 'clean', '--all'], {cwd: projectRoot});
+        },
+      },
+    ],
+    metro: [
+      {
+        label: 'Clean Metro cache',
+        action: () => cleanDir(`${os.tmpdir()}/metro-*`),
+      },
+      {
+        label: 'Clean Haste cache',
+        action: () => cleanDir(`${os.tmpdir()}/haste-map-*`),
+      },
+      {
+        label: 'Clean React Native cache',
+        action: () => cleanDir(`${os.tmpdir()}/react-*`),
+      },
+    ],
+    npm: [
+      {
+        label: 'Remove node_modules',
+        action: () => cleanDir(`${projectRoot}/node_modules`),
+      },
+      ...(verifyCache
+        ? [
+            {
+              label: 'Verify npm cache',
+              action: async () => {
+                await execa('npm', ['cache', 'verify'], {cwd: projectRoot});
+              },
+            },
+          ]
+        : []),
+    ],
+    watchman: [
+      {
+        label: 'Stop Watchman',
+        action: async () => {
+          await execa(
+            os.platform() === 'win32' ? 'tskill' : 'killall',
+            ['watchman'],
+            {cwd: projectRoot},
+          );
+        },
+      },
+      {
+        label: 'Delete Watchman cache',
+        action: async () => {
+          await execa('watchman', ['watch-del-all'], {cwd: projectRoot});
```

Comment:
I wonder, shouldn't we call `watchman watch-del ${cwd}`? watch-del-all will clear all the caches for unrelated projects (e.g. Jest uses it), and usually it's about this specific cwd.

Reply:
Another issue is that Watchman will freeze and use 100% CPU when you're switching between projects. I don't know what the cause is, but killing watchman and resetting the cache is the only thing that works.

Reply:
oh, I didn't take monorepos into account. Let's keep it as is then